#! /bin/sh /usr/share/dpatch/dpatch-run
## 30xenchangeset.dpatch by Guido Trotter <ultrotter@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Make the changeset configurable, defaulting to the old value

@DPATCH@
diff -urNad xen-3.0.1+hg8743/xen/Makefile /tmp/dpep.Ek3jgb/xen-3.0.1+hg8743/xen/Makefile
--- xen-3.0.1+hg8743/xen/Makefile	2006-02-18 19:57:57.971524368 +0100
+++ /tmp/dpep.Ek3jgb/xen-3.0.1+hg8743/xen/Makefile	2006-02-19 19:20:15.000000000 +0100
@@ -8,6 +8,7 @@
 export XEN_SUBVERSION    = 0
 export XEN_EXTRAVERSION  = .1
 export XEN_FULLVERSION   = $(XEN_VERSION).$(XEN_SUBVERSION)$(XEN_EXTRAVERSION)
+export XEN_CHANGESET	?= $(shell ((hg parents || head -n 7 ../ChangeLog || echo date: unavailable) | awk '{FS="changeset:[ ]+"}/^changeset/{CS=$$2};{FS="date:[ ]+"}/^date/{D=$$2}; END {print D, CS}') 2>/dev/null)
 
 export BASEDIR          := $(CURDIR)
 
@@ -96,7 +97,7 @@
 	    -e 's/@@version@@/$(XEN_VERSION)/g' \
 	    -e 's/@@subversion@@/$(XEN_SUBVERSION)/g' \
 	    -e 's/@@extraversion@@/$(XEN_EXTRAVERSION)/g' \
-	    -e 's!@@changeset@@!$(shell ((hg parents || head -n 7 ../ChangeLog || echo date: unavailable) | awk '{FS="changeset:[ ]+"}/^changeset/{CS=$$2};{FS="date:[ ]+"}/^date/{D=$$2}; END {print D, CS}') 2>/dev/null)!g' \
+	    -e 's/@@changeset@@/$(XEN_CHANGESET)/g' \
 	    < include/xen/compile.h.in > $@.new
 	@cat include/xen/banner.h >> $@.new
 	@mv -f $@.new $@
